codegen/smithy-go-codegen/src/test/java/software/amazon/smithy/go/codegen/GoMiddlewareGeneratorTest.java

```diff
@@ -0,0 +1,27 @@
+package software.amazon.smithy.go.codegen;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+
+import org.junit.jupiter.api.Test;
+
+public class GoMiddlewareGeneratorTest {
+    @Test
+    public void generatesSerializeMiddlewareDefinition() {
+        GoWriter writer = new GoWriter("middlewaregentest");
+
+        GoMiddlewareGenerator.newSerializeMiddleware("someMiddlewareId")
+                .writeMiddleware(writer, (m, w) -> {
+                    w.openBlock("return next.$L(ctx, in)", m.getHandleMethodName());
+                });
+
+        String generated = writer.toString();
+
+        assertThat(generated, containsString("type someMiddlewareId struct {"));
```

Comment:
You don't need to get rid of this, but these sorts of tests aren't super useful. Better to test the generated code than to make assertions about what the output looks like. Granted that's more or less impossible until we have a complete vertical slice.

Reply:
Agreed not a worthwhile test once we have other functioning components.